lib/LANraragi/Utils/Plugins.pm

```diff
@@ -114,25 +114,44 @@ sub get_plugin_parameters {
     my $plugin   = get_plugin($namespace);
     my %pluginfo = $plugin->plugin_info();
 
-    my @args = ();
+    my %args;
 
-    # Fill with default values first
-    foreach my $param ( @{ $pluginfo{parameters} } ) {
-        push( @args, $param->{default_value} );
-    }
+    if ( ref( $pluginfo{parameters} ) eq 'ARRAY' ) {
 
-    # Replace with saved values if they exist
-    if ( $redis->hexists( $namerds, "enabled" ) ) {
-        my $argsjson = $redis->hget( $namerds, "customargs" );
-        $argsjson = redis_decode($argsjson);
+        my @args;
 
-        #Decode it to an array for proper use
-        if ($argsjson) {
-            @args = @{ decode_json($argsjson) };
+        # Fill with default values first
+        foreach my $param ( @{ $pluginfo{parameters} } ) {
+            push( @args, $param->{default_value} );
         }
-    }
+
+        # Replace with saved values if they exist
+        if ( $redis->hexists( $namerds, "enabled" ) ) {
+            my $argsjson = $redis->hget( $namerds, "customargs" );
+            $argsjson = redis_decode($argsjson);
+
+            #Decode it to an array for proper use
+            if ($argsjson) {
+                @args = @{ decode_json($argsjson) };
+            }
+        }
+        $args{customargs} = \@args;
+
+    } elsif ( ref( $pluginfo{parameters} ) eq 'HASH' ) {
+
+        my %params = $redis->hgetall($namerds);
+
+        # TODO: param conversion block, remove after deprecation period
+        if ( $pluginfo{to_named_params} && exists $params{customargs} ) {
+            %params = convert_to_named_params_and_persist( $redis, $namerds, $pluginfo{to_named_params}, %params );
+        }
+
+        %args = map { $_ => redis_decode( $params{$_} ) } keys %params;
+
+    }    # else { # should we die here? }
```

Comment:
Don't see why you'd need to die here, parameter-less plugins can still exist.  

Although we might want to log a warning on the if above when `customargs` still exists yet the plugin doesn't have a `to_named_params` conversion method written, telling the user that the plugin is malformed and old settings might be lost.